source/client/HandView.py

```diff
@@ -56,13 +56,20 @@ def update(self):
         if self.controller._state.round == -1:
             self.mesgBetweenRounds(self.betweenrounds)
             if self.round_advance:
-                self.round_index = self.round_index + 1
-                if self.round_index < len(Meld_Threshold):
-                    self.betweenrounds[0] = 'This is the round of ' + str(Meld_Threshold[self.round_index]) + ' ! '
+                self.next_round = self.next_round + 1
+                if self.next_round < len(Meld_Threshold):
+                    self.betweenrounds[0] = 'This is the round of ' + str(Meld_Threshold[self.next_round]) + ' ! '
                 else:
                     self.betweenrounds = ['Game has concluded. Scores for each round can be found in command window.']
                 self.round_advance = False
         else:
+            if not self.next_round == self.controller._state.round:
```

Comment:
I think this makes more sense to do entirely in the controller in Network_deal method, that would make more sense since the controller's state is what tracks score history anywah.

Reply:
Once scores are calculated and transmitted to server, the controller forgets them.  Therefore, to know how many rounds a player has missed, the player needs public info readily available in tableView..  Hence the padding is done in TableView.